src/overlays/actors/ovl_Obj_Wind/z_obj_wind.c

```diff
@@ -28,21 +32,139 @@ ActorInit Obj_Wind_InitVars = {
     (ActorFunc)ObjWind_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B24480[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 4000, ICHAIN_STOP),
 };
 
-#endif
+void ObjWind_Init(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    s32 pad;
+    WaterBox* waterbox;
+    f32 ySurface;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if (WaterBox_GetSurface1(play, &play->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &ySurface,
+                             &waterbox) &&
+        (this->actor.world.pos.y < ySurface)) {
+        this->isUnderWater = 1;
+    }
+}
+
+void ObjWind_Destroy(Actor* thisx, PlayState* play) {
+}
+
+typedef struct ObjWindStruct {
+    /* 0x0 */ s16 unk_0;
+    /* 0x2 */ s16 unk_2;
+    /* 0x4 */ s16 unk_4;
+    /* 0x6 */ s16 unk_6;
+    /* 0x8 */ s16 unk_8;
+} ObjWindStruct;
+
+static ObjWindStruct D_80B2448C[] = {
+    { 530, 90, 3000, 2000, 0 },   { 1120, 100, 2000, 1000, 0 }, { 620, 100, 2000, 1000, 0 },
+    { 1620, 100, 2000, 1000, 0 }, { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },  { 400, 100, 2000, 1000, 0 },
+    { 400, 100, 2000, 1000, 0 },  { 600, 120, 3000, 2000, 0 },  { 500, 80, 3000, 2000, 0 },
+    { 400, 60, 2000, 2000, 0 },   { 470, 70, 2000, 1000, 0 },
+};
+
+static s32 D_80B245CC = -1;
+
+void ObjWind_Update(Actor* thisx, PlayState* play) {
+    ObjWind* this = (ObjWind*)thisx;
+    ObjWindStruct* entry = &D_80B2448C[OBJ_WIND_GET_TYPE(thisx)];
+    Player* player;
+    f32 cosYTimesX;
+    f32 cosX;
+    f32 sinYTimesX;
+    Vec3f posCopy;
+    Vec3f nearestPoint;
+    Vec3f sp54;
+    f32 dist;
```

Comment:
```suggestion
    f32 dist; // reused temp
```
Kind of annoying but the stack seems really full, so I haven't been able to find something better